src/components/fx/hover.js

```diff
@@ -464,47 +492,28 @@ function _hover(gd, evt, subplot, noHoverEvent) {
     }
 
     function selectClosestPoint(pointsData, spikedistance) {
-        if(!pointsData.length) return null;
-        var resultPoint;
-        var pointsDistances = pointsData.map(function(point, index) {
-            var xa = point.xa,
-                ya = point.ya,
-                xpx = xa.c2p(xval),
-                ypx = ya.c2p(yval),
-                dxy = function(point) {
-                    var rad = point.kink,
-                        dx = (point.x1 + point.x0) / 2 - xpx,
-                        dy = (point.y1 + point.y0) / 2 - ypx;
-                    return Math.max(Math.sqrt(dx * dx + dy * dy) - rad, 1 - 3 / rad);
-                };
```

Comment:
We're not replicating the distance function here, because that only works for `scatter` anyhow. Now we're letting each `hoverPoints` module tell us both a `distance` (based on `hovermode`) and a `spikeDistance` (always as if in `closest` mode, but may be calculated differently or omitted if that hover label should not make a spike line)